src/cheri_ptw.sail

```diff
@@ -24,14 +24,21 @@ function ptw_error_to_str(e) =
 
 overload to_str = {ptw_error_to_str}
 
+function ext_get_ptw_error(ext_ptw : ext_ptw) -> PTW_Error =
+  match (ext_ptw) {
+    PTW_CAP_OK        => PTW_No_Permission(),
+    PTW_LOAD_CAP_ERR  => PTW_Ext_Error(AT_CAP_ERR),
+    PTW_STORE_CAP_ERR => PTW_Ext_Error(AT_CAP_ERR)
```

Comment:
Why have two PTW errors, conflate them down to one PTW_Ext_Error, and then pry them back apart in `translationException`?

Reply:
You're right, currently we should have just the one PTE_CAP_ERR/PTW_CAP_ERR, that's more in keeping with the base model. But AMOSWAP.C changes that if we go with your suggestion of returning _true_ for SC violations and letting them through, since then the code needs to be able to distinguish between LC and SC being missing so it knows whether to trap or tag clear. Thus I think this is right; we keep the state about the details of the page table walk in ext_ptw, and when we need to trap we report a generic trap that's then turned into the right one based on the _instruction_ (because AMOs always trap with store/AMO faults, even if it was the load part of it that caused the trap).

Reply:
> even if it was the load part of it that caused the trap

That's... not going to be fantastic for the load generation counter story (though of course we can hook both 0x1A and 0x1B in software for CLG checks).  I'd prefer we deviated from upstream here, but I'm open to more authoritative opinions.

Reply:
I assume the reasoning is that your store fault handler normally does a strict superset of what your load fault handler does, since if you can store to it you can read from it (W-but-not-R is a reserved state I believe), so you might as well take the full fault up-front and do everything in one go rather than do the bare minimum for a load and then take store fault upon retry. My view is we should follow upstream, and then if it proves to be as painful as we fear for revocation, work out what the right approach is, but only then. It might turn out to not be expensive in practice.

Reply:
Yea, in practice we shouldn't be doing too many AMOSWAP.C-s against CLG-mismatching pages, one hopes, and adding a test for CLG-mismatch to the SAMO handler should be cheap because SAMO's on the way to CoW most of the time, right?